src/NHSD.BuyingCatalogue.Identity.Api/Repositories/UserRepository.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using NHSD.BuyingCatalogue.Identity.Api.Models;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.Repositories
+{
+    public sealed class UserRepository : IUserRepository
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+
+        public UserRepository(UserManager<ApplicationUser> userManager)
+        {
+            _userManager = userManager ?? throw new ArgumentNullException(nameof(userManager));
+        }
+
+        public async Task<ApplicationUser> FindByIdAsync(string userId) => await _userManager.FindByIdAsync(userId);
```

Comment:
This isn't really something I'd change, but it's a bit weird to me how we await userManager here, and then the caller has to await us awaiting the userManager. Functionally it seems like we could just return _userManager.FindByIdAsync without awaiting it, and that would save on one whole await call, letting the caller await the _userManager itself.

I think that would probably go against some sort of coding practice, but it does feel a bit odd to me when we're essentially nesting awaits for no reason.

Reply:
Good point & you are correct. Most guidance I have seen suggests awaiting here rather than just returning the task. Quick look up found the following resource:

[Async await advice](https://blog.stephencleary.com/2016/12/eliding-async-await.html)